src/main/java/io/opentracing/contrib/spanmanager/DefaultSpanManager.java

```diff
@@ -0,0 +1,109 @@
+package io.opentracing.contrib.spanmanager;
+
+import io.opentracing.NoopSpan;
+import io.opentracing.Span;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Default {@link SpanManager} implementation using {@link ThreadLocal} storage,
+ * maintaining a stack-like structure of linked managed spans.
+ * <p>
+ * The linked managed spans provide the following stack unwinding algorithm:
+ * <ol>
+ * <li>If the released span is not the <em>managed</em> span, the <em>current managed</em> span is left alone.</li>
+ * <li>Otherwise, the first parent that is <em>not yet released</em> is set as the new managed span.</li>
+ * <li>If no managed parents remain, the <em>managed span</em> is cleared.</li>
+ * <li>Consecutive <code>release()</code> calls for already-released spans will be ignored.</li>
+ * </ol>
+ */
+public final class DefaultSpanManager implements SpanManager {
+
+    private static final Logger LOGGER = Logger.getLogger(DefaultSpanManager.class.getName());
+
+    private static final DefaultSpanManager INSTANCE = new DefaultSpanManager();
```

Comment:
For a library I prefer the static method as it allows future flexibility.
I agree that in this case it's doubtful it will be needed.